public/tutorials/using-infura-custom-provider.md

```diff
@@ -1,136 +1,152 @@
-# Using Infura (or any custom provider)
+# Using Infura (or a custom provider)
 
-[Infura](https://infura.io/) is a hosted Ethereum node cluster that lets your users run your application without requiring them to set up their own Ethereum node or wallet. For security reasons Infura does not manage users' private keys, which means they cannot sign transactions on your behalf (or your users' behalf). Similarly, Truffle doesn't intuitively know how to sign your transactions -- it usually hands that task over to the Ethereum node -- so if you'd like to deploy your contracts using Infura, you must first tell Truffle how you'd like these transactions signed.
+[Infura](https://infura.io/) is a hosted Ethereum node cluster that lets your users run your application without requiring them to set up their own Ethereum node or wallet.
+![Infura](/tutorials/images/infura/infura.png)
 
-Fortunately libraries can help us along. In this example we'll use an [hd wallet mnemonic](https://en.bitcoin.it/wiki/Deterministic_wallet) to generate our list of wallets, and we'll use the [web3-provider-engine](https://github.com/MetaMask/provider-engine) to provide us with built-in signing functionality.
+You may not be familiar with Infura, but if you've used MetaMask you've used Infura, as it is the node that MetaMask connects to.
 
-### Getting Started
+For security reasons Infura does not manage your private keys, which means Infura cannot sign transactions on your behalf.
 
-First install the required dependencies:
+However, Truffle can sign transactions through the use of its `HDWalletProvider`. This provider can handle the transaction signing as well as the connection to the Ethereum network. ([Read more about HDWalletProvider](https://github.com/trufflesuite/truffle-hdwallet-provider).)
 
-```
-$ npm install ethereumjs-wallet bip39 web3-provider-engine web3
+This tutorial will show you how to use Infura to migrate an existing dapp to an Ethereum network (in this case, Ropsten). This tutorial will assume that you already have a dapp to migrate. If you want a test dapp, feel free to use our [Pet Shop](/tutorials/pet-shop) tutorial dapp.
+
+## Install HDWalletProvider
+
+Truffle's `HDWalletProvider` is a separate npm package:
+
+```shell
+npm install truffle-hdwallet-provider
 ```
 
-We recommend you run the above command with the `--save` flag to save this configuration in your project's `package.json` file.
```

Comment:
I'm assuming you took this out because newer versions of npm autosave?

Reply:
I think I thought it was part of the now-unneeded content. Should I put it back?

Reply:
Nah, npm will automatically save now if greater than 5.0 or something. It's still helpful for those using npm < 5.0 though.